doc/doxygen/CMakeLists.txt

```diff
@@ -34,17 +34,25 @@ IF(NOT DOXYGEN_FOUND)
     )
 ENDIF()
 
+############################################################################
 #
-# A glob. I'm sorry
+# Process the tutorial and code-gallery files into inputs for doxygen
 #
-file(GLOB DEAL_II_STEPS
-  ${CMAKE_SOURCE_DIR}/examples/step-*
-  )
+# The tutorial/CMakeLists.txt file is also responsible for building
+# the top-level tutorial pages that may link to the code gallery,
+# so we need to have the code gallery stuff in place first.
+#
+############################################################################
+
+ADD_SUBDIRECTORY(code-gallery)
```

Comment:
The `CMakeLists.txt` that I want to include here is the one that sets up the various internal targets to process the code gallery into the deal.II documentation. It is really something that I need to do here, rather than in the code-gallery itself, in the same way as we process the tutorial in `doc/doxygen/tutorial/CMakeLists.txt` rather than `examples/CMakeLists.txt`. So I'll leave the `ADD_DIRECTORY` here unconditionally, but will add the check whether the top-level directory exists to `doc/doxygen/code-gallery/CMakeLists.txt`.